modules/silicon_design/scripts/build/notebooks/examples/renode/tuning.md

```diff
@@ -0,0 +1,188 @@
+staging_bucket = 'foss-fpga-tools-ext-antmicro-radlab-silicon-renode-staging'
+project = 'foss-fpga-tools-ext-antmicro'
+location = 'us-east4'
+machine_type = 'n2-standard-4'
+notebook = pathlib.Path('renode.ipynb')
+now = datetime.datetime.now().strftime('%Y%m%d%H%M%S')
+prefix = notebook.stem
+staging_dir = f'{prefix}-tuning-{now}'
+staging_dir
+```
+
+## Delete data from previous runs and stage the notebook for the experiment
+
+```python tags=[]
+!rm -r {staging_dir}
+!gsutil rm -r gs://{staging_bucket}/{staging_dir}/
+!gsutil cp {notebook} gs://{staging_bucket}/{staging_dir}/{notebook}
+```
+
+## Create Parameters and Metrics specs
+
+We want to find the best value for *target density* and *die area* in order optimize *total power* consumption.
+
+Those keys map to the [parameters](https://papermill.readthedocs.io/en/latest/usage-parameterize.html) and [metrics](https://github.com/GoogleCloudPlatform/cloudml-hypertune) advertised by the notebook.
+
+```python tags=[]
+from google.cloud.aiplatform import hyperparameter_tuning as hpt
+
+parameter_spec = {
+    'quantum': hpt.DoubleParameterSpec(min=0.0001, max=0.4, scale='linear'),
+    'mbs': hpt.IntegerParameterSpec(min=10000, max=100000, scale='linear'),
+    'mips': htp.IntegerParameterSpec(min=10, max=1000, scale='linear')
+}
+
+metric_spec={'execution_time': 'minimize'}
+```
+
+## Create Custom Job spec
+
+```python tags=[]
+from google.cloud import aiplatform
+import pathlib
+
+worker_pool_specs = [{
+    'machine_spec': {
+        'machine_type': machine_type,
+    },
+    'replica_count': 1,
+    'container_spec': {
+        'image_uri': worker_image,
+        'args': ['/usr/local/bin/papermill-launcher', 
+                 f'gs://{staging_bucket}/{staging_dir}/{notebook}',
+                 f'$AIP_MODEL_DIR/{prefix}_out.ipynb',
+                 '--runs_dir=/tmp']
+    }
+}]
+custom_job = aiplatform.CustomJob(display_name=f'{prefix}-custom-job',
+                                  worker_pool_specs=worker_pool_specs,
+                                  staging_bucket=staging_bucket,
+                                  base_output_dir=f'gs://{staging_bucket}/{staging_dir}')
+```
+
+## Run Hyperparameter tuning job
+
+```python tags=[]
+from google.cloud import aiplatform
+
+parameters_count = len(parameter_spec.keys())
+metrics_count = len(metric_spec.keys())
+max_trial_count = 100 * parameters_count * metrics_count
+parallel_trial_count = int(max_trial_count / 20)
+
+hpt_job = aiplatform.HyperparameterTuningJob(
+    display_name=f'{prefix}-tuning-job',
+    custom_job=custom_job,
+    metric_spec=metric_spec,
+    parameter_spec=parameter_spec,
+    max_trial_count=max_trial_count,
+    parallel_trial_count=parallel_trial_count,
+    max_failed_trial_count=max_trial_count,
+    location=location)
+hpt_job.run(sync=True)
+```
+
+## Check the current state
```

Comment:
Maybe split that in a different notebook? So that it's easier to do the analysis concurrently.